Mathlib/Topology/Algebra/ContinuousMonoidHom.lean

```diff
@@ -413,3 +415,278 @@ theorem locallyCompactSpace_of_hasBasis (V : ℕ → Set Y)
+@[to_additive "The type of two-sided continuous isomorphism of additive groups."]
+structure ContinuousMulEquiv [Mul G] [Mul H] extends MulEquiv G H , Homeomorph G H
+
+/-- The Homeomorphism induced from a two-sided continuous isomorphism of groups. -/
+add_decl_doc ContinuousMulEquiv.toHomeomorph
+
+/-- The Homeomorphism induced from a two-sided continuous isomorphism additive groups. -/
+add_decl_doc ContinuousAddEquiv.toHomeomorph
+
+/-- Notation for a `ContinuousMulEquiv`. -/
+infixl:25 " ≃ₜ* " => ContinuousMulEquiv
+
+/-- Notation for an `ContinuousAddEquiv`. -/
+infixl:25 " ≃ₜ+ " => ContinuousAddEquiv
+
+section
+
+namespace ContinuousMulEquiv
+
+variable {M N : Type*} [TopologicalSpace M] [TopologicalSpace N] [Mul M] [Mul N]
+
+section coe
+
+@[to_additive]
+instance : EquivLike (M ≃ₜ* N) M N where
+  coe f := f.toFun
+  inv f := f.invFun
+  left_inv f := f.left_inv
+  right_inv f := f.right_inv
+  coe_injective' f g h₁ h₂ := by
+    cases f
+    cases g
+    congr
+    exact MulEquiv.ext_iff.mpr (congrFun h₁)
+
+@[to_additive]
+instance : MulEquivClass (M ≃ₜ* N) M N where
+  map_mul f := f.map_mul'
+
+@[to_additive]
+instance : HomeomorphClass (M ≃ₜ* N) M N where
+  map_continuous f := f.continuous_toFun
+  inv_continuous f := f.continuous_invFun
+
+@[to_additive] -- shortcut instance that doesn't generate any subgoals
+instance : CoeFun (M ≃ₜ* N) fun _ ↦ M → N where
+  coe f := f
+
+/-- Two continuous multiplicative isomorphisms agree if they are defined by the
+same underlying function. -/
+@[to_additive (attr := ext)
+  "Two continuous additive isomorphisms agree if they are defined by the same underlying function."]
+theorem ext {f g : M ≃ₜ* N} (h : ∀ x, f x = g x) : f = g :=
+  DFunLike.ext f g h
+
+@[to_additive]
+protected theorem congr_arg {f : M ≃ₜ* N} {x x' : M} : x = x' → f x = f x' :=
+  DFunLike.congr_arg f
+
+@[to_additive]
+protected theorem congr_fun {f g : M ≃ₜ* N} (h : f = g) (x : M) : f x = g x :=
+  DFunLike.congr_fun h x
+
+@[to_additive (attr := simp)]
+theorem coe_mk (f : M ≃* N) (hf1 : Continuous f.toFun) (hf2 : Continuous f.invFun) :
+    (mk f hf1 hf2 : M → N) = f := rfl
+
+@[to_additive]
+theorem toEquiv_eq_coe (f : M ≃ₜ* N) : f.toEquiv = f :=
+  rfl
+
+/-- Makes a continuous multiplicative isomorphism from
+a homeomorphism which preserves multiplication. -/
+@[to_additive "Makes an continuous additive isomorphism from
+a homeomorphism which preserves addition."]
+def mk' (f : M ≃ₜ N) (h : ∀ x y, f (x * y) = f x * f y) : M ≃ₜ* N :=
+  ⟨⟨f.toEquiv,h⟩, f.continuous_toFun, f.continuous_invFun⟩
+
+end coe
+
+section bijective
+
+@[to_additive]
+protected theorem bijective (e : M ≃ₜ* N) : Function.Bijective e :=
+  EquivLike.bijective e
+
+@[to_additive]
+protected theorem injective (e : M ≃ₜ* N) : Function.Injective e :=
+  EquivLike.injective e
+
+@[to_additive]
+protected theorem surjective (e : M ≃ₜ* N) : Function.Surjective e :=
+  EquivLike.surjective e
+
+-- Porting note (#10618): `simp` can prove this
```

Comment:
Why are you making a porting note? This is surely not code ported from mathlib3?

Reply:
Sorry that I directly copied everything from `MulEquiv`, I should delete it.